src/winerror.cr

```diff
@@ -0,0 +1,2181 @@
+class WinError < Errno
+  # NOTE: `get_last_error` must be called BEFORE an instance of this class
+  # is malloced as it would change the "last error" to SUCCESS
+  def self.new(message)
+    new(message, LibC.get_last_error)
+  end
+
+  def initialize(message, code)
+    buffer = uninitialized UInt8[256]
+    size = LibC.format_message(LibC::FORMAT_MESSAGE_FROM_SYSTEM, nil, code, 0, buffer, buffer.size, nil)
+    details = String.new(buffer.to_unsafe, size).strip
+    super "#{message}: [WinError #{code}, #{details}]", winerror_to_errno(code)
+  end
+
+  def winerror_to_errno(winerror)
+    case winerror
+    when    2; return 2
```

Comment:
There is a undocumented function _dosmaperr. Python [uses this functions](https://github.com/python/cpython/blob/master/PC/generrmap.c) to generate [a mapper](https://github.com/python/cpython/blob/master/PC/errmap.h). Because the mapper has not be changed for 6 years, I supposed it's quite correct and just coped those numbers.